### ISSUE TITLE
Add rule for preventing stray "set -x"

### ISSUE THREAD

**Issue by ZeroEcks** [koalaman/shellcheck#2416]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
set -x

SECRET="secret-here"

use-secret $SECRET
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

Do not use debug mode

#### Reasoning

I would like to ensure that developers do not commit `set -x` to repositories, to prevent the accidental leaking of secrets from shell scripts, debug logging such as `set -x` should also not be committed in general imo. There can be a flag to allow it perhaps?

**Comment by david-lev** [comment]
I don't understand what that'll give you. After all, it'll always be possible to run `bash -x your-script.sh` and activate the debug mode.

**Comment by ZeroEcks** [comment]
The issue I am trying to prevent, is a way to fail CI builds if developers accidentally commit `set -x` to a deployment script which may include variables such as `AWS_ACCESS_KEY` or similar secrets as environment variables. This would lead to secrets being logged.

You also generally, would not want to commit debug statements to repositories in my experience?

The argument they could run it as `bash -x script.sh` doesn't apply since CI wouldn't do that (unless on purpose, but then they would be acting maliciously)

I think it would be beneficial to have to acknowledge that you are committing debug statements that could potentially expose environment variables, since you can always ignore the warning for specific lines.

**Comment by TinCanTech** [comment]
Does `shellcheck` review commits now ? :black_cat:

**Comment by ZeroEcks** [comment]
It's not too much of a stretch to imagine that people run linting tools as part of their pipeline, and before committing code? We can probably safely assume most code written ends up in a repository?

I think the same would apply if your just writing a script, it should be preferred to run things explicitly at runtime with `bash -x script.sh` rather than leaving debugging messages in final scripts (potentially leading to leaking secrets)

Especially since most scripts I write personally will contain at least one secret environment variable.

**Comment by TinCanTech** [comment]
I do see _exactly_ what you are saying .. just not sure which side of the fence i fall.
